cve_bin_tool/util.py

```diff
@@ -280,24 +280,21 @@ def __missing__(self, key: str) -> list[CVE] | set[str]:
 
 def regex_find(
     lines: str, version_patterns: list[Pattern[str]], ignore: list[Pattern[str]]
-) -> list[str]:
+) -> set[str]:
     """Search a set of lines to find all matches for the given regex"""
-    versions = list()
+    versions = set()
 
     for pattern in version_patterns:
         version_matches = pattern.finditer(lines)
         for match in version_matches:
-            # before collecting a potential version number, ensure the version string isn't on the ignore list
-            if not check_ignored(match.string, ignore):
+            matched_text = match.group(0)
```

Comment:
Let's preserve the original comment here, so future contributors can understand this function more easily.